src/inference/src/ie_core.cpp

```diff
@@ -2083,8 +2083,10 @@ void Core::add_extension(const ie::IExtensionPtr& extension) {
 
 void Core::add_extension(const std::string& library_path) {
     try {
-        add_extension(ov::detail::load_extensions(library_path));
-    } catch (const std::runtime_error&) {
+        const std::string absolute_path = ov::util::get_absolute_file_path(library_path);
```

Comment:
The real problem of the bug is that `dlopen` doesn't treat `libname.so` as relative path (but `./libname.so` will be), and searches it in ENV and system dirs. If you try to convert it to absolute path, you prohibit searching in ENV with such implementation

I suggest to handle `libname.so` as relative path first (converted to absolute), and if file not found, try to pass it as is to `dlopen`.
Note that `ov::util::get_absolute_file_path` throws exception if file doesn't exist (on Linux, but not on Windows)

Reply:
Changed